packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -3442,6 +3446,14 @@ export function attach(
       createIsPathAllowed('state', null),
     );
 
+    // Ref is a special case;
+    // don't dehydrate it in the same way (because it's not hydratable/inspectable).
+    // Just stringify it instead...
```

Comment:
How is `ref` different than a prop holding a ref? For example, before `React.forwardRef` we often had a custom prop like `forwardedRef`.

Reply:
This is a good question. Ref is different (at least currently) because it has special React semantics. But you're right, a custom prop name like `forwardedRef` would be inspectable and would show Fiber internals. Maybe we need a more robust solution for filtering keys like that on the DevTools side. 🤔 Could be a follow up to consider.

Reply:
Also to consider: The `ref` might not point to a DOM element but point to imperative methods or other mutable values.

Reply:
My gut feeling was that stringifying is still okay in that case, but maybe...I don't know. I don't really know if I believe in this change overall yet.